### DISCUSSION TITLE
Character controllers and thread-safe usage of TempAllocator

### DISCUSSION THREAD

**Discussion by zopsicle** [jrouwe/JoltPhysics#1240] [Q&A] [answered]
The documentation for `TempAllocator` reads:

> Note that allocations and frees can take place from different threads, but the order is guaranteed though job dependencies, so it is not needed to use any form of locking.

`CharacterVirtual::ExtendedUpdate` and `CharacterVirtual::SetShape` each take a `TempAllocator` but are called outside of `PhysicsSystem::Update`, and therefore not involved with jobs.

Does this mean I need to externally synchronize access to the `TempAllocator` when using character controllers?

**Comment by jrouwe** [comment, marked as answer]
Yes, if you update multiple characters at the same time you need to provide your own synchronization (we use a thread local temp allocator for this)
